src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityAPIBookingService.kt

```diff
@@ -0,0 +1,92 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import mu.KLogging
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import org.springframework.web.util.UriComponentsBuilder
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.CommunityAPIClient
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ActionPlanAppointment
+import java.time.LocalDate
+import java.time.LocalTime
+import java.time.OffsetDateTime
+import javax.validation.constraints.NotNull
+
+@Service
+class CommunityAPIBookingService(
+  @Value("\${appointments.bookings.enabled}") private val bookingsEnabled: Boolean,
+  @Value("\${interventions-ui.baseurl}") private val interventionsUIBaseURL: String,
+  @Value("\${interventions-ui.locations.view-appointment}") private val interventionsUIViewAppointment: String,
+  @Value("\${community-api.locations.book-appointment}") private val communityApiBookAppointmentLocation: String,
+  @Value("\${community-api.appointments.office-location}") private val officeLocation: String,
+  @Value("\${community-api.integration-context}") private val integrationContext: String,
+  val communityAPIClient: CommunityAPIClient,
+) {
+  companion object : KLogging()
+
+  fun book(existingAppointment: ActionPlanAppointment, appointmentTime: OffsetDateTime?, durationInMinutes: Int?) {
+    when (bookingsEnabled) {
```

Comment:
nit: would prefer guard clause to avoid extra level of indentation\

i.e. if (!bookingsEnabled) return

Reply:
Will do